Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -84,6 +84,8 @@ type AuthInfo struct {
 	ClientKeyData []byte `json:"client-key-data,omitempty"`
 	// Token is the bearer token for authentication to the kubernetes cluster.
 	Token string `json:"token,omitempty"`
+	// ActAs is the username to act-as.
+	ActAs string `json:"act-as,omitempty"`
```

Comment:
> Or were you thinking of the controller having many clients?

I was thinking of the controller having a special kind of client that could set this as a one off.  I didn't think that we'd want to change this per-request in the common case, because I wouldn't expect most client calls to attempt to act-as a different user.